system/Cache/Handlers/BaseHandler.php

```diff
@@ -14,12 +14,65 @@
 use Closure;
 use CodeIgniter\Cache\CacheInterface;
 use Exception;
+use InvalidArgumentException;
 
 /**
  * Base class for cache handling
  */
 abstract class BaseHandler implements CacheInterface
 {
+	/**
+	 * Reserved characters that cannot be used in a key or tag.
+	 *
+	 * @see https://github.com/symfony/cache-contracts/blob/c0446463729b89dd4fa62e9aeecc80287323615d/ItemInterface.php#L43
```

Comment:
Just read this line: `but MUST be able to return the original unmodified key string.` Since MD5 hashing is a one-way process, then aren't we deviating again from PSR-6?

Reply:
That link is because I copied this code from Symfony, more of an attribution than a reference. Maybe I should remove the `@see` and just have the link?

`BaseHandler::validateKey()` is not trying to be PSR-6 compliant - we'd have bigger issues to address if that were the case. The PSR-6 adapter will use this but handle its own keys and exceptions.